Lib/Inc/CFIFOBuffer.h

```diff
@@ -26,6 +28,7 @@ class CFIFOBuffer
     bool put(T data)
     {
         bool b_success = false;
+        __disable_irq();
```

Comment:
This definitely needs comments. Explain what is happening and why you are doing that. Otherwise, someone might "refactor it to optimise bandwidth".